src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -2265,7 +2265,7 @@ private void handleStateBootreplacing(InetAddress newNode, String[] pieces)
      *
      * @param endpoint node
      */
-    private synchronized void handleStateNormal(final InetAddress endpoint, final String status)
+    private void handleStateNormal(final InetAddress endpoint, final String status)
```

Comment:
Which specific data structures was the `synchronized` intending to guard reads/writes? Are there specific JMM effects that will no longer hold with removal?

Reply:
I believe the `synchronized` was added as a "why not be even safer" during code review, not during testing on RCs (see: https://github.com/palantir/cassandra/pull/336/files#r858486528). The primary intention as I understand was guarding reads/writes to TokenMetadata during node IP replacements, which are still in place (cc @Sam-Kramer).

Reply:
Got it, thanks!